.github/workflows/manual-upgrade-testnet-l2.yml

```diff
@@ -0,0 +1,149 @@
+          echo "RESOURCE_TAG_NAME=devtestnetlatest" >> $GITHUB_ENV
+          echo "RESOURCE_STARTING_NAME=D" >> $GITHUB_ENV
+          echo "RESOURCE_TESTNET_NAME=devtestnet" >> $GITHUB_ENV
+
+      - name: 'Output env vars'
+        id: outputVars
+        run: |
+          echo "L2_ENCLAVE_DOCKER_BUILD_TAG=${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_HOST_DOCKER_BUILD_TAG=${{env.L2_HOST_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG=${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG=${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_TAG_NAME=${{env.RESOURCE_TAG_NAME}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_STARTING_NAME=${{env.RESOURCE_STARTING_NAME}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_TESTNET_NAME=${{env.RESOURCE_TESTNET_NAME}}" >> $GITHUB_OUTPUT
+
+      - name: 'Login to Azure docker registry'
+        uses: azure/docker-login@v1
+        with:
+          login-server: testnetobscuronet.azurecr.io
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: 'Build and push obscuro node images'
+        run: |
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/enclave.Dockerfile  .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/host.Dockerfile .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}} -f testnet/contractdeployer.Dockerfile .
+          docker push ${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}} -f testnet/hardhatdeployer.Dockerfile .
+          docker push ${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}}
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        host_id: [ 0,1 ]
+
+    steps:
+      - name: 'Extract branch name'
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF_NAME})"
+        id: extract_branch
+
+      - name: 'Login via Azure CLI'
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: 'Update Obscuro node-${{ matrix.host_id }} on Azure'
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            az vm run-command invoke -g Testnet -n "${{needs.build.outputs.RESOURCE_STARTING_NAME}}-${{ matrix.host_id }}-${{ github.event.inputs.github_deploy_number }}"  \
+            --command-id RunShellScript \
+            --scripts '
+              cd /home/obscuro/go-obscuro/testnet/ \
+              docker compose down host enclave && \
+              docker compose pull && \
+              docker compose up host enclave
```

Comment:
How does this differentiate between testnet and dev-testnet? i.e. without giving the name does this not always use the docker-compose. yml file?

Reply:
Good point. Fixing this

Reply:
Fixed by removing the `docker-compose.dev-testnet.yml` file.
The `.env` file should decide the params of the node execution including what images to use.

I think this even makes it neater in terms of file usage :) Good catch!